### PR TITLE
KFSPTS-30090: Improve configuration of DummyAntiVirusServiceImpl

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/krad/antivirus/service/impl/DummyAntiVirusServiceImpl.java
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/krad/antivirus/service/impl/DummyAntiVirusServiceImpl.java:21]: I wonder if this setup could be simplified if we keep the existing constructors and add a String argument constructor that calls this constructor?
That may allow us to leave DummyScanResult and the unit tests as they are for the most part.
Thoughts?

Comment (nkimble) [reply src/main/java/edu/cornell/kfs/krad/antivirus/service/impl/DummyAntiVirusServiceImpl.java:21]: I think that these changes make the "dummy" service more closely mimic the actual ClamAV implementation of these services (i.e. reading a value from a properties file and having the ScanResult implementation be responsible for keeping that string "result" value in sync with the "status" enum value) which IMHO is what we want the unit tests to exercise.
I did try your suggestion and ran into issues.

**Review by ajd299 — CHANGES_REQUESTED** [review]
I agree that it would be preferable to use the enum as @cah292 suggested if it can be automatically convert to the String type, however if you are trying to mimic the ClamAV setup that makes sense too.

File: src/main/java/edu/cornell/kfs/krad/antivirus/util/DummyAVUtils.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/krad/antivirus/util/DummyAVUtils.java:18]: StringUtils.defaultString is null safe and returns an empty String should the input parameter to the static method (i.e. result) happen to be a null.  
I will have the default dummy implementation return value be Status.ERROR (similar to the real implementation) instead of null. Thanks for catching that.

**Review by nkimble — COMMENTED** [review]
If the rest of the team is ok with it, I would prefer to keep the dummy code as close to the real code implementation as possible and not use the enums at this time.

**Review by cah292 — COMMENTED** [review]
The revised changes look good to me. This can be merged after Jay and Andrew have reviewed and approved it.

**Comment by jhulslander** [comment]
I am fine with moving forward with these changes, I did have one idea for improvement, but I don't think we have to do it.

File: src/main/java/edu/cornell/kfs/krad/antivirus/service/impl/DummyScanResult.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/krad/antivirus/service/impl/DummyScanResult.java:46]: Another alternative would be to use `Status.valueOf(result)`, and use a `catch` block to handle cases where the status name doesn't line up with an enum constant.

**Review by ajd299 — APPROVED** [review]
Code approved, merging to develop

**Pull request merged**
